### PR TITLE
Add router CA certificate to the client

### PR DESCRIPTION
**Pull request by mkumatag**
Fixes: https://github.com/openshift/origin/issues/24079

### CODE REVIEW

**Comment by ironcladlou** [comment]
/cc @Miciah 

This is weird. I'm not sure things should be reaching into the openshift-ingress-operator namespace. I thought openshift-config-managed was the location of shared information like this. @Miciah, what do you think?

**Comment by Miciah** [comment]
> This is weird. I'm not sure things should be reaching into the openshift-ingress-operator namespace. I thought openshift-config-managed was the location of shared information like this. @Miciah, what do you think?

Agreed.  Please use the `router-ca` configmap in the `openshift-config-managed` namespace.

**Comment by bparees** [comment]
/assign @ironcladlou @stlaz 
/unassign @adambkaplan 

(more appropriate owners given what this touches)

**Comment by mkumatag** [comment]
> > This is weird. I'm not sure things should be reaching into the openshift-ingress-operator namespace. I thought openshift-config-managed was the location of shared information like this. @Miciah, what do you think?
> 
> Agreed. Please use the `router-ca` configmap in the `openshift-config-managed` namespace.

Thanks, I will do the required changes. :)

**Comment by mkumatag** [comment]
> > This is weird. I'm not sure things should be reaching into the openshift-ingress-operator namespace. I thought openshift-config-managed was the location of shared information like this. @Miciah, what do you think?
> 
> Agreed. Please use the `router-ca` configmap in the `openshift-config-managed` namespace.

@ironcladlou @Miciah I have done the required changes, can you please review the code again?

**Comment by stlaz** [comment]
There is no commit message and no bug/issue reference - what problem is this trying to solve?

**Comment by mkumatag** [comment]
> There is no commit message and no bug/issue reference - what problem is this trying to solve?

Here is the issue: https://github.com/openshift/origin/issues/24079 mentioned in the first comment.

**Comment by stlaz** [comment]
yeh, sorry, I realized that and removed that comment right after that. However, in my environment, when I do: `openshift-tests run all --dry-run | grep -E "Feature:OAuthServer" | grep -E "OAuth server" | openshift-tests run all -f -`, the test passes. Maybe it's not an issue of the test?

File: test/extended/oauth/expiration.go [hunk 1]
```diff
@@ -114,7 +114,8 @@ func testTokenFlow(oc *util.CLI, newRequestTokenOptions oauthserver.NewRequestTo
 }
 
 func testCodeFlow(oc *util.CLI, newRequestTokenOptions oauthserver.NewRequestTokenOptionsFunc, client *oauthv1.OAuthClient, expectedExpiresIn int32) {
-	anonymousClientConfig := rest.AnonymousClientConfig(oc.AdminConfig())
+	anonymousClientConfig, err := oauthserver.InjectRouterCA(oc, rest.AnonymousClientConfig(oc.AdminConfig()))
```
Comment (stlaz) [inline test/extended/oauth/expiration.go]: `oc.AdminConfig()` should already have all the certs required

Comment (mkumatag) [reply test/extended/oauth/expiration.go]: seems like my kubeconfig itself does not have the certificate, as you mentioned I need to go back and check installer is not injecting the router CA!

**Pull request closed without merging**